lib/controllers/users.js

```diff
@@ -0,0 +1,178 @@
+'use strict';
+const boom = require('boom');
+const joi = require('joi');
+const _ = require('lodash');
+const moment = require('moment');
+
+const internals = {};
+
+/**
+ * Test whether a user with at least one of the properties exists
+ * @param  {object} authUtils an instance of the authentication utilities
+ * @param  {object} userModel bookshelf model to fetch user data
+ * @param  {object} userData  object containing properties to match the user on
+ * @return {object}          undefined if no matches found,
+ *                                 otherwise a bookshelf object
+ */
+internals.getUser = (authUtils, UserModel, userData) => {
+    const user = new UserModel();
+    const formatQuery = user._utils.formatQuery;
+
+    //clone to avoid mutation
+    let username = userData.username;
+    let qbObj = {orWhere: formatQuery(userData)};
+    if (username) {
+        qbObj.orWhere.username = authUtils.encryptUsername(username);
+    }
+
+    return user
+        .query(qbObj)
+        .fetch();
+};
+
+internals.defaultValues = {
+    siteId: '-1',
+    activeFlag: 'Y',
+    passwordExpDate: moment().add(1, 'years').format('MM/DD/YYYY'),
+    acctExpDate: moment().add(1, 'years').format('MM/DD/YYYY')
+};
+
+/**
+ * Call the toJSON method on the object
+ * @param  {Object} obj object to call the toJSON method on (bookshelf)
+ * @return {string}     JSON
+ */
+internals.callToJson = (obj) => {
+    return obj.toJSON();
+};
+
+/**
+ * Call the fetch method on the object
+ * @param  {Object} obj object to call the fetch method on (bookshelf)
+ * @return {Promise}
+ */
+internals.callFetch = (obj) => {
+    //remove all attributes but username
+    obj.attributes = { username: obj.get('username') };
+    return obj.fetch();
+};
+
+/**
+ * hide protected fields so that they are no returned
+ * @param  {object} obj bookshelf object of user
+ * @return {object}     sanitized model
+ */
+internals.sanitize = (obj) => {
```

Comment:
Agreed, but (eventually), we could just decorate `User` to have a `User.sanitize()` method, which we would manually call before replying...

Out of scope for this, but good to keep in mind.